File: Arrays & Strings/Roman To Integer.kt
class Solution {
    fun romanToInt(s: String): Int {
        val romanMap = mapOf('I' to 1, 'V' to 5, 'X' to 10, 'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000)
        var sum = 0
        var prevValue = romanMap[s[0]] ?: 0

        for (i in 1 until s.length) {
            val currentValue = romanMap[s[i]] ?: 0
            sum += if (currentValue > prevValue) {
                -prevValue
            } else {
                prevValue
            }
            prevValue = currentValue
        }
        sum += prevValue

        return sum
    }
}

File: Arrays & Strings/Roman to Integer/Roman To Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        std::unordered_map<char, int> romanMap = {
            {'I', 1},   {'V', 5},   {'X', 10},  {'L', 50},
            {'C', 100}, {'D', 500}, {'M', 1000}};

        int result = 0;

        for (unsigned int i = 0; i < s.size(); ++i) {
            if (i + 1 < s.size() && romanMap[s[i + 1]] > romanMap[s[i]]) {
                result += romanMap[s[i + 1]] - romanMap[s[i]];
                ++i;
            } else {
                result += romanMap[s[i]];
            }
        }

        return result;
    }
};
